docs/decisions/0008-application_state_differenciation.rst

```diff
@@ -0,0 +1,104 @@
+8. Intentional separation between local application state and API data state using ``@tanstack/react-query``
+============================================================================================================
+
+Status
+******
+
+Accepted  (January 2024)
+
+Context
+*******
+
+The ``frontend-app-admin-portal`` currently implements ``@tanstack/react-query`` predominantly within the ``learner-credit-management`` feature. ``@tanstack/react-query`` is a tool used to manage server state in react applications.
+
+See `0006-tanstack-react-query ADR <https://github.com/openedx/frontend-app-admin-portal/blob/master/docs/decisions/0006-tanstack-react-query.rst>`_ for more information
+
+As part of its implementation, a pattern has emerged of API data from the react-query cache is being populated within context providers. This has resulted in the unintended consequence of page refreshes when the top-level component context data is updated with the latest data from the ``react-query`` cache.
+
+The re-fetch behavior of ``react-query`` results in page components that aren't dependent on the updated context data to reload.
+
+The re-fetch behavior was occurring in the background of a component while the user is interacting with the page or if they had returned focus back onto the browser while interacting with other tabs or applications.
+
+An example where populating context data from ``react-query`` cache data resulted in unintentional behavior is the ``EnterpriseSubsidiesContext`` containing API data from the query key 'learner-credit-management.budgets', ``isFetchingBudgets`` attribute which has since been fixed.
+
+While the user was interacting with a ``FullscreenModal`` component and had not completed their intended activity within the modal before the global ``staleTime`` had expired, the re-fetch behavior would trigger resulting in a full page re-load, thus closing the modal without saving the users state information (form data etc.).
+
+Decisions
+*********
+
+We will take a more intentional approach when implementing API data fetching using the ``react-query`` in the following ways:
+
+#. Avoid adding API data fetched using ``react-query`` to a context layer
+    * If a context layer does require API data to exist in the context layer, do not use ``react-query`` to retrieve the data
+#. When using ``react-query`` make the call as close to the component that uses the returned response data
```

Comment:
[clarification] Is this item getting at how we should simply rely on calling the custom hook wrapping `useQuery` where relevant in the component tree as opposed to only calling it once higher up in the component tree and sharing its result with context?

That is, if you need API data, call ``useQuery`` directly (generally, regardless of where in the component tree), as opposed to trying to expose query data via some other mechanism besides the ``useQuery`` cache?

Perhaps this could be re-worded a bit to make this a bit more clear? That is, no need to expose API data/state in context layer because the ``useQuery`` hook can be called numerous times during rendering without risking duplicate API calls.

I also have a similar question as Brian's below around not fully understanding what is meant by "may require mapping a component from the parent component which completes the API call".